fix a serious bug when get next overlapped matched base in the case t…

wulj2 opened the pull request:
…he match base is in just the first one in the stretch of match and this stretch of match is after some other stretch of non-match sequences(such as ins, del, .etc)
fixed#1195

valeriuo commented:
Thank your for the fix. However, I've made [one](https://github.com/valeriuo/htslib/commit/2d034837bff3ebd9cba6318a9ace585d2a0bbdb9) which is simpler. Instead of keeping track of a previous matching segment, it resets `icig` to -1 every time, so that the first base of each matching segment starts from 0.
I rebased and squashed your commits and added my changes on top of yours, to preserve your authorship.

wulj2 commented:
wow, that is a really clever fix, I love it so much.

On Thu, Jan 14, 2021 at 12:37 AM Valeriu Ohan <notifications@github.com>
wrote:

> Thank your for the fix. However, I've made one
> <https://github.com/valeriuo/htslib/commit/2d034837bff3ebd9cba6318a9ace585d2a0bbdb9>
> which is simpler. Instead of keeping track of a previous matching segment,
> it resets icig to -1 every time, so that the first base of each matching
> segment starts from 0.
> I rebased and squashed your commits and added my changes on top of yours,
> to preserve your authorship.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/samtools/htslib/pull/1196#issuecomment-759569913>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJBMPDAM7234H73R2B6DHODSZXD23ANCNFSM4VHHORTA>
> .
>

wulj2 commented:
[Elegantly fixed](https://github.com/valeriuo/htslib/commit/2d034837bff3ebd9cba6318a9ace585d2a0bbdb9) by @valeriuo
